### ISSUE TITLE
Semantic Highlighting issues with Type Hints for Generic Types

### ISSUE THREAD

**Issue by hassanselim0** [microsoft/pylance-release#459]
## Environment data
-   Language Server version: 2020.9.8
-   OS and version: Windows 10 v2004
-   Python version: Python 3.8.2

## Expected behaviour

Given this function definition:
```python
def calc(nums: List[int]):
    pass
```
The entire `List[int]` span should have the `typeHint` semantic modifier.

## Actual behaviour

(note: I've overridden the token color for `*.typeHint` to red for clarity)
![image](https://user-images.githubusercontent.com/1431770/95121141-ea056e00-074e-11eb-9c77-92a751bea20b.png)
`List` gets the `variable` semantic without the `typeHint` modifier.

![image](https://user-images.githubusercontent.com/1431770/95121215-06a1a600-074f-11eb-9587-3a4c2382ed90.png)
The square brackets don't get any semantic token or modifier.

Edit: It seems like these issues happen on class variable annotations too, not just in function argument annotations.

## Code Snippet / Additional information

Related Feature Request: #335

**Comment by hassanselim0** [comment]
Just found something that could narrow down this issue:
![image](https://user-images.githubusercontent.com/1431770/95199826-bde4fe00-07dd-11eb-9c14-74eedc133d97.png)
The semantic token type for all these types is `class`, except for `Dict` which is `variable`. So the problem could be that the `typeHint` modifier currently only applies to classes and not variables, even though variables can contain type objects.

**Comment by huguesv** [comment]
Thanks for the report. We should be able to fix those.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2020.10.0, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/master/CHANGELOG.md#2020100-7-october-2020

**Comment by hassanselim0** [comment]
Thank you, I just updated and the problem is fixed.
The square brackets still don't get any token though, which I think means they can't get the `typeHint` modifier.
I'm not sure if it would be worth the effort though. My main purpose with the original feature request is to make the type hints less eye-catching instead of standing out (currently trying our faded colors), to try and keep the code as readable as possible.
For me, what we have now is a good step and I would consider myself satisfied.

For reference, these are the colors customizations I'm experimenting right now with the Amethyst Dark theme:
![image](https://user-images.githubusercontent.com/1431770/95655103-52649e80-0b05-11eb-8635-110ca1e8cca9.png)
It's part of [Pydantic](https://github.com/samuelcolvin/pydantic) Model, a typing library that I've started using a lot in the past month.
